lib/src/SortableList/SortableList.tsx

```diff
@@ -0,0 +1,265 @@
+import React from "react";
+import classnames from "classnames";
+import { SortableItemProps } from "./SortableItem";
+import "./dragtouch.polyfills";
+import "./sortable-list.scss";
+
+const dragAndDropIcon: JSX.Element = (
+    <svg width="10px" fill="currentColor" height="16px" viewBox="0 0 10 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
+        <g id="Symbols" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
+            <g id="Group" fill="currentColor">
+                <circle id="Oval" cx="2" cy="2" r="2" />
+                <circle id="Oval-Copy-2" cx="2" cy="8" r="2" />
+                <circle id="Oval-Copy-4" cx="2" cy="14" r="2" />
+                <circle id="Oval-Copy" cx="8" cy="2" r="2" />
+                <circle id="Oval-Copy-3" cx="8" cy="8" r="2" />
+                <circle id="Oval-Copy-5" cx="8" cy="14" r="2" />
+            </g>
+        </g>
+    </svg>
+);
+
+export type SortableListProps = JSX.IntrinsicElements["div"] & {
+    /** on sort callback */
+    onSort: (list: Array<string>) => void;
+    /** disable sorting */
+    disabled?: boolean;
+};
+
+type PositionDelta = {
+    x: number;
+    y: number;
+};
+
+type OrderItem = Pick<SortableItemWrapperProps, "disabled" | "uniqueKey">;
+
+/** The component allows for sorting list by drag and drop. */
+export const SortableList: React.FC<SortableListProps> = ({ onSort, className, disabled, ...props }: React.PropsWithChildren<SortableListProps>) => {
+    const dragContainerRef = React.useRef<HTMLDivElement>();
+    const [defaultOrders, setDefaultOrders] = React.useState<OrderItem[]>([]);
+    const [draggingOrders, setDraggingOrders] = React.useState<OrderItem[]>(defaultOrders);
+    const [currentItemNode, setCurrentItemNode] = React.useState<HTMLDivElement>(null);
+    const [currentItem, setCurrentItem] = React.useState<number>(null);
```

Comment:
notice the item type is number, is this referring to the item index? if it is, maybe we can refactor the variable name to `currentItemIndex`? _(it is slightly longer, but it allows reader to understand at a glance that it is referring to the index of the item)_